networkit/cpp/graph/test/GraphToolsGTest.cpp

```diff
@@ -222,6 +222,40 @@ TEST_P(GraphToolsGTest, testRandomNode) {
     }
 }
 
+TEST_P(GraphToolsGTest, testRandomNodes) {
+    constexpr count n = 20;
+    constexpr double p = 0.01;
+
+    auto G = ErdosRenyiGenerator(n, p, directed()).generate();
+    if (weighted()) {
+        G = Graph(G, true, G.isDirected());
+    }
+
+    count sampleAll = G.numberOfNodes();
+    std::vector<node> allSample = GraphTools::randomNodes(G, sampleAll);
+    EXPECT_EQ(allSample.size(), sampleAll);
+    for (count i = 0; i < sampleAll; i++) {
+        for (count j = i + 1; j < sampleAll; j++)
+            EXPECT_NE(allSample[i], allSample[j]);
+    }
```

Comment:
Just use `EXPECT_EQ(std::unordered_set<node>{allSample.begin(), allSample.end()}.size(), sampleAll);`, same for the cases below.